### PR TITLE
refactor: let library to support “@”

### PR DESCRIPTION
**Pull request by Eusen**
When I execute `ng g application @apps/app1`, I get the following results:
``` text
- @app/
-       app1/
(newProjectRoot = "")
```

But !!! When I execute `ng g library @libs/lib1`, I got the result:
``` text
- libs/
-      lib1/
```

This is too awkward !!!

Please! Remove the `.replace(/^@/, '')` here, or add the `.replace(/^@/, '')` to the application!

I think it's really weird now...

### CODE REVIEW

**Comment by splincode** [comment]
@alan-agius4 is it still active?

**Comment by dgp1130** [comment]
Going through some old PRs and took a quick look at this. We looked back through history and found the original motivation for stripping `@` symbols to be [terminal autocomplete issues](https://github.com/angular/angular-cli/issues/10172#issuecomment-384445140). This still seems to exist at least with Bash on Mac, where `@` directories can't be autocompleted. The CLI also generates a path mapping in `tsconfig.json`, so it isn't quite this trivial of a change.

This is probably a case where the cure is worse than the disease. Changing this behavior now would introduce a discrepancy for all new generated libraries (which include `@` in the directory), while all old library don't include `@`. We would probably need an option for existing workspaces, but we don't feel a change like this is worth supporting such an option. The autocomplete issue is also not obvious to users and it's hard to surface this as a decision with a clear cost/benefit. If we were implementing this feature now we might be more inclined to include `@` in the directory, but the nature of having existing files which don't have the `@` changes the calculus significantly.

The workaround is also fairly straightforward, since you just need to rename the directory. Arguably someone could make a custom schematic that wrapped `ng generate` and automatically renamed the output, though I'm not sure how tricky that would be.

TL;DR: The original motivation for stripping the `@` still exists, removing it would break existing user workflows, and the workaround is a simple file rename. As a result, we think it's better to keep the status quo rather than introduce additional friction with minimal value to users.
